src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -106,6 +106,8 @@ public static Runnable combine(Runnable... runnables) {
 
     private static final Pattern UNDERSCORE_TO_SPACE = Pattern.compile("_");
 
+    public static final Pattern VALID_NUMERIC = Pattern.compile(".[0-9]*");
```

Comment:
There is mistake as it now allows any char on first position.
```suggestion
    public static final Pattern VALID_NUMERIC = Pattern.compile("[0-9]*");
```

This is only quick fix so it can be merged now. Correct way would be to use already existing https://github.com/GregTechCEu/GregTech/blob/master/src/main/java/gregtech/api/gui/widgets/TextFieldWidget2.java#L29

Reply:
I didn't know there were Pattern consts in TextFieldWidget2, so I've switched to them now and removed the unneeded Pattern in GTUtility (https://github.com/GregTechCEu/GregTech/pull/1448/commits/69a540a53bfe0bca5e24c22a6e6dec4027a0dac3)

Reply:
Good job! And don't worry about it, it is not obvious enough place to have them.